progressbar/utils.py

```diff
@@ -384,21 +384,22 @@ def format_time(time, precision=datetime.timedelta(seconds=1)):
     '''
     precision_seconds = precision.total_seconds()
 
-    if isinstance(time, six.basestring + six.numeric_types):
+    if isinstance(tval, six.string_types + six.integer_types + (float, )):
         try:
-            time = datetime.timedelta(seconds=six.long_int(time))
+            castfunc = long if six.PY2 else int  # NOQA
```

Comment:
Might be easier to simply use: `castfunc = six.integer_types[-1]`

Reply:
Not sure if six guarantees the order, have to reread that.

Reply:
Not sure if it strictly does, but the tuple hasn't changed in 7 years so it seems pretty stable :)
https://github.com/benjaminp/six/blame/master/six.py#L50

Reply:
Alternatively... `type(2**64)` ;)